blog/2022-08-05-paralus-audit-logs/index.md

```diff
@@ -0,0 +1,90 @@
+---
+slug: paralus-audit-logs
+title: "How To Use Audit Logs In Paralus"
+authors: [atul]
+tags: [paralus]
+---
+
+The concept of safety and security have been prevalent from the early days of human existence. Back in the day, watch towers and sentries were a way for kings to keep a watch on their surroundings.
+
+In the modern day setup, information security is a bigger challenge than physical security. Confidential & sensitive data runs into petabytes and it becomes all the more important to safeguard not only the data but the entire infrastructure.
+
+<!--truncate-->
+
+Businesses today employ an array of security solutions to safeguard their infrastructure. A common and vital element of any security strategy/tool is logging.
+
+Logs are digital, text based records of all the activities that happen in an organization. These provide you with an audit trail that you can use to monitor various activities within your organization, identify policy violations and any unusual activity.
+
+Logs can be generated by various systems in your environment - applications, firewalls, network devices and the likes. These are a great way to know what is happening/happened, who did it, when did it happen, where did it happen. These details play a critical role during compliance audits.
+
+## Logging in Kubernetes
+
+Kubernetes as we all know is the most widely used container orchestration with a lot of moving parts. Nodes, clusters, pods, services, replica sets and a lot of other things.
+
+Logging activities across all these elements isn't an easy task, especially due to its ephemeral and dynamic nature.  With such a nature, Kubernetes provides three different types of logs:
+
+1. **Application logs:** Logs from apps deployed in clusters that help understand what's happening in the app.
+2. **Kubernetes Cluster components:** Logs from API-server, etcd, kube-proxy etc. that help troubleshoot cluster issues.
+3. **Kubernetes audit logs:** Logs related to API activity primarily used for audit & governance purposes.
+
+In this post we'll focus on Kubernetes Audit logs.
+
+Kubernetes provides a way to keep audit logs with a sequence of events that take place in a cluster. These are generated by users, applications or the control plane itself. Audit logs are generated for every request that generates an audit event and processed according to an audit policy.
```

Comment:
There is no direct relation between k8s audit logs and paralus audit logs. We are not directly leveraging anything inside k8s for logs. Maybe we can skip this entire section?
/cc @niravparikh05

Reply:
So how is Paralus audit log is different from K8s audit log? I was looking for a link between K8s logs and Paralus's audit logs.

Reply:
Paralus and k8s audit logs are two different things. k8s audit logs is something that k8s provides so as to let you collect audit logs. You will have to take these logs own your on and push somewhere.

Now when it comes to paralus, the k8s logs that you get are for the requests that go through relay. We collect these in the system and push to es and make it available in the UI.

Reply:
Fixed